server/routes.go

```diff
@@ -841,6 +841,25 @@ func (s *Server) HeadBlobHandler(c *gin.Context) {
 }
 
 func (s *Server) CreateBlobHandler(c *gin.Context) {
+	ib, ok := intermediateBlobs.Load(c.Param("digest"))
+	if ok {
+		p, err := GetBlobsPath(ib.(string))
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
+
+		if _, err := os.Stat(p); errors.Is(err, os.ErrNotExist) {
```

Comment:
you should probably include a debug statement in here for the cache hit but something else removed the storage.